crypto/bn/bn_rsa_fips186_4.c

```diff
@@ -228,8 +228,8 @@ int ossl_bn_rsa_fips186_4_gen_prob_primes(BIGNUM *p, BIGNUM *Xpout,
     if (!bn_rsa_fips186_4_find_aux_prob_prime(Xp1i, p1i, ctx, rounds, cb)
             || !bn_rsa_fips186_4_find_aux_prob_prime(Xp2i, p2i, ctx, rounds, cb))
         goto err;
-    /* (Table B.1) auxiliary prime Max length check */
-    if ((BN_num_bits(p1i) + BN_num_bits(p2i)) >=
+    /* (Table A.1) auxiliary prime Max length check */
```

Comment:
This file seems to mix references in the comments to FIPS 186-4 and FIPS 186-5 (and the name of the file is consequently quite confusing). Can we explicitly reference 186-5 here:

```suggestion
    /* (FIPS 186-5 Table A.1) auxiliary prime Max length check */
```